k4FWCore/components/PodioInput.cpp

```diff
@@ -184,14 +184,15 @@ PodioInput::PodioInput(const std::string& name, ISvcLocator* svcLoc) : Consumer(
     error() << "Could not get PodioDataSvc" << endmsg;
   }
   fillReaders();
-
-  auto key = edm4hep::EventHeaderName;
-  if (std::find(m_collectionNames.begin(), m_collectionNames.end(), key) == m_collectionNames.end()) {
-    m_collectionNames.value().push_back(key);
-  }
 }
 
 void PodioInput::operator()() const {
+  if (m_podioDataSvc->hasCollection(edm4hep::EventHeaderName)) {
```

Comment:
Since it only happens one the `hasCollection` function can be removed, I think this should work:

```suggestion
  if (m_podioDataSvc->getEventFrame.get(edm4hep::EventHeaderName)) {
```

Reply:
Done.